### PR TITLE
refactor: BEM convention for modules

### PR DESCRIPTION
**Pull request by attiyaIshaque**
#### Description:

* Implemented BEM convention for register and forgot password page 
* Moved the custom classes into a separate files

Tickets:
https://2u-internal.atlassian.net/browse/VAN-1421
https://2u-internal.atlassian.net/browse/VAN-1424

#### How Has This Been Tested?

It has been tested locally

### CODE REVIEW

File: src/sass/_registration.scss [hunk 1]
```diff
@@ -1,3 +1,3 @@
-.register-stateful-button-width {
+.stateful--button--width {
```
Comment (zainab-amir) [inline src/sass/_registration.scss:1]:
```suggestion
.register-button {
```

File: src/sass/_forgot_password.scss [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// Forgot Password Page
+.forgot-password-button-width {
```
Comment (zainab-amir) [inline src/sass/_forgot_password.scss:2]: Avoid giving css properties in class name
```suggestion
.forgot-password-button {
```

File: src/sass/_forgot_password.scss [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+  min-width: 6rem;
+}
```
Comment (zainab-amir) [inline src/sass/_forgot_password.scss:4]: End of file missing

File: src/sass/_forgot_password.scss [hunk 4]
```diff
@@ -0,0 +1,4 @@
+// Forgot Password Page
```
Comment (zainab-amir) [inline src/sass/_forgot_password.scss:1]: We do not need this comment

**Review by zainab-amir — COMMENTED** [review]
Also this is missing register page classes
- suggested-username-close-button
- opt-checkbox
- email-error-alert
- username-suggestion
- scroll-suggested-username
- username-suggestion-label
- email-warning-alert-link

There can be others as well and some of these might already be following BEM convention but most of these still need to be moved to the `_registration.scss` file.

**Comment by attiyaIshaque** [comment]
@zainab-amir I have addressed the comments.

File: src/sass/_registration.scss [hunk 5]
```diff
@@ -1,7 +1,91 @@
...
+    color: $primary !important;
+
+    &:hover {
+      text-decoration: underline;
+      color: $info-700 !important;
+    }
+  }
+}
+
+.email-warning-alert-link {
```
Comment (zainab-amir) [inline src/sass/_registration.scss:35]:
```suggestion
.email-suggestion-alert-warning {
```

**Review by zainab-amir — COMMENTED** [review]
The classes have been moved to the registration specific scss file but the BEM convention has not been applied to it.

File: src/sass/_registration.scss [hunk 6]
```diff
@@ -1,7 +1,91 @@
...
+  }
+
+  .alert-link {
+    color: $primary !important;
+
+    &:hover {
+      text-decoration: underline;
+      color: $info-700 !important;
+    }
+  }
```
Comment (zainab-amir) [inline src/sass/_registration.scss:32]:
```suggestion
..email-suggestion-alert-error {
  padding: 0.5rem 1rem;

  .email-suggestion__close {
    float: right;

    &:hover {
      cursor: pointer;
    }
  }

  .email-suggestion__text {
    font-size: 0.75rem;
    line-height: 1.25rem;
  }

  .email-suggestion__link {
    color: $primary !important;

    &:hover {
      text-decoration: underline;
      color: $info-700 !important;
    }
  }
```

**Comment by attiyaIshaque** [comment]
@zainab-amir I have addressed all the comments.

File: src/sass/_registration.scss [hunk 7]
```diff
@@ -1,7 +1,91 @@
...
+}
+
+.suggested-username__error {
+  position: relative;
+  margin-top: -13.7%;
+  margin-bottom: 11%;
+  margin-left: 15px;
+}
+
+.username-scroll-suggested--form-field {
```
Comment (zainab-amir) [inline src/sass/_registration.scss:86]:
```suggestion
.username-suggestions--field--scrollable {
```

File: src/sass/_registration.scss [hunk 8]
```diff
@@ -1,7 +1,91 @@
...
+
+.form-field--checkbox {
+  .pgn__form-label {
+    font-size: 0.75rem;
+    line-height: 1.25rem;
+  }
+  margin-left: 3px;
+}
+
+.username-suggestion--button {
```
Comment (zainab-amir) [inline src/sass/_registration.scss:52]:
```suggestion
.username-suggestions--chip {
```

**Review by zainab-amir — COMMENTED** [review]
Let's use `username-suggestions` as block for all classes related to `suggested-username` and `username-suggestion`.

**Review by zainab-amir — APPROVED** [review]
Test the design on stage before deploying on prod

**Pull request merged**